### ISSUE TITLE
"Cannot read property 'default' of undefined" import error

### ISSUE THREAD

**Issue by timothy-bailey-redbox** [storybookjs/storybook#13021]
**Describe the bug**
I have been having some trouble with getting storybook to display some of my components. Specifically there appears to be 2 files in my application that don't get imported correctly by storybook/webpack. (But are fine in the actual application build). 

The error message in the console I see is `Unexpected error: TypeError: Cannot read property 'default' of undefined` in Chrome, and `Unexpected error: TypeError: _adminReducer__WEBPACK_IMPORTED_MODULE_15__ is undefined` in Firefox. "adminReducer" being the name of one of the 2 files that doesn't import.

I'm not sure if it is relevent but both files are parts of my redux reducer. But it is only 2 of my 16 reducers, the other 14 appear to function fine. If I comment out the 2 reducers that don't work then all of my stories show up no problem. Those 2 reducers also don't contain any code that is syntatically different to the others. I even replaced their entire contents with just a signle function that returns an empty object and it still fails to import.

I realise it's slightly odd that I'm wrapping my stories in my redux store, but some of my components need to have certain parent elements that requre the store to be running to function correctly.

**To Reproduce**
I'm unsure how to replicate this issue outside of my codebase as I don't understand what has caused these 2 files to fail to import.

**Expected behavior**
I expect all of my imports to load and function normally as they do in the application.

**Screenshots**
N/A

**Code snippets**
My combined reducer looks like this:
```javascript
import { connectRouter, LocationChangeAction } from 'connected-react-router';
import { History } from 'history';
import { State } from './domain';
import metaReducer from "./metaReducer";
import userReducer from "./userReducer";
import accountReducer from "./accountReducer";
import campaignDataReducer from "./campaignDataReducer";
import dashboardReducer from "./dashboardReducer";
import signupReducer from "./signupReducer";
import campaignCreationReducer from "./campaignCreationReducer";
import takeOwnershipReducer from "./takeOwnershipReducer";
import campaignPageReducer from "./campaignPageReducer";
import campaignLocationReducer from "./campaignLocationReducer";
import campaignAttributionReducer from "./campaignAttributionReducer";
import campaignBudgetReducer from "./campaignBudgetReducer";
import { ActionType } from "../actions/ActionType";
import campaignKeywordReducer from "./campaignKeywordReducer";
import campaignTargetingReducer from './campaignTargetingReducer';
import adminReducer from './adminReducer';

export default function createReducer(history: History<any>, initialState = {} as State): (state: State, action: ActionType) => State {
    const routerReducer = connectRouter(history);
    return (state = initialState, action: ActionType) => ({
        router: routerReducer(state.router, action as unknown as LocationChangeAction),
        meta: metaReducer(state.meta, action, state),
        user: userReducer(state.user, action, state),
        account: accountReducer(state.account, action, state),
        admin: adminReducer(state.admin, action, state),
        campaignData: campaignDataReducer(state.campaignData, action, state),
        dashboard: dashboardReducer(state.dashboard, action, state),
        signup: signupReducer(state.signup, action, state),
        campaignCreation: campaignCreationReducer(state.campaignCreation, action, state),
        takeOwnership: takeOwnershipReducer(state.takeOwnership, action, state),
        campaignPage: campaignPageReducer(state.campaignPage, action, state),
        campaignLocationPage: campaignLocationReducer(state.campaignLocationPage, action, state),
        campaignAttributionPage: campaignAttributionReducer(state.campaignAttributionPage, action, state),
        campaignBudgetPage: campaignBudgetReducer(state.campaignBudgetPage, action, state),
        campaignKeywordPage: campaignKeywordReducer(state.campaignKeywordPage, action, state),
        campaignTargetingPage: campaignTargetingReducer(state.campaignTargetingPage, action, state),
    });
}
```
And to get my components stories showing I just comment out the 2 reducers that have the issue:
```javascript
import { connectRouter, LocationChangeAction } from 'connected-react-router';
import { History } from 'history';
import { State } from './domain';
import metaReducer from "./metaReducer";
import userReducer from "./userReducer";
import accountReducer from "./accountReducer";
import campaignDataReducer from "./campaignDataReducer";
import dashboardReducer from "./dashboardReducer";
import signupReducer from "./signupReducer";
import campaignCreationReducer from "./campaignCreationReducer";
import takeOwnershipReducer from "./takeOwnershipReducer";
import campaignPageReducer from "./campaignPageReducer";
import campaignLocationReducer from "./campaignLocationReducer";
import campaignAttributionReducer from "./campaignAttributionReducer";
import campaignBudgetReducer from "./campaignBudgetReducer";
import { ActionType } from "../actions/ActionType";
import campaignKeywordReducer from "./campaignKeywordReducer";
import campaignTargetingReducer from './campaignTargetingReducer';
import adminReducer from './adminReducer';

export default function createReducer(history: History<any>, initialState = {} as State): (state: State, action: ActionType) => State {
    const routerReducer = connectRouter(history);
    return (state = initialState, action: ActionType) => ({
        router: routerReducer(state.router, action as unknown as LocationChangeAction),
        meta: metaReducer(state.meta, action, state),
        user: userReducer(state.user, action, state),
        account: accountReducer(state.account, action, state),
        // admin: adminReducer(state.admin, action, state),
        campaignData: campaignDataReducer(state.campaignData, action, state),
        dashboard: dashboardReducer(state.dashboard, action, state),
        signup: signupReducer(state.signup, action, state),
        // campaignCreation: campaignCreationReducer(state.campaignCreation, action, state),
        takeOwnership: takeOwnershipReducer(state.takeOwnership, action, state),
        campaignPage: campaignPageReducer(state.campaignPage, action, state),
        campaignLocationPage: campaignLocationReducer(state.campaignLocationPage, action, state),
        campaignAttributionPage: campaignAttributionReducer(state.campaignAttributionPage, action, state),
        campaignBudgetPage: campaignBudgetReducer(state.campaignBudgetPage, action, state),
        campaignKeywordPage: campaignKeywordReducer(state.campaignKeywordPage, action, state),
        campaignTargetingPage: campaignTargetingReducer(state.campaignTargetingPage, action, state),
    });
}
```

My webpack config in the .storybook directory looks like:
```javascript
const webpack = require('webpack');
const getClientEnvironment = require('../config/env');
const env = getClientEnvironment('');

module.exports = async ({ config }) => {
    config.module.rules.push({
        test: /\.(js|mjs|jsx|ts|tsx)$/,
        loader: require.resolve('babel-loader'),
        options: {
          customize: require.resolve(
            'babel-preset-react-app/webpack-overrides'
          ),
          plugins: [
            "@babel/plugin-proposal-optional-chaining",
            "@babel/plugin-proposal-nullish-coalescing-operator",
            [
              require.resolve('babel-plugin-named-asset-import'),
              {
                loaderMap: {
                  svg: {
                    ReactComponent:
                      '@svgr/webpack?-svgo,+titleProp,+ref![path]',
                  },
                },
              },
            ],
          ],
          cacheDirectory: true,
          cacheCompression: false,
          compact: false,
        },
    });
    config.resolve.extensions.push('.ts', '.tsx', '.js', '.jsx');

    config.module.rules.push({
        test: /\.css/,
        loader: require.resolve('postcss-loader'),
        options: {
            ident: 'postcss',
            plugins: () => [
                require('postcss-flexbugs-fixes'),
                require('postcss-preset-env')({
                    autoprefixer: {
                        flexbox: 'no-2009',
                    },
                    features: {
                        'nesting-rules': true,
                        'color-mod-function': true,
                    },
                    stage: 3,
                }),
                require('postcss-normalize'),
            ],
        }
    });
    config.resolve.extensions.push('.css');

    config.plugins.push(new webpack.DefinePlugin(env.stringified));
    return config;
};
```

This is one of my stories that impliments the reducer and doesn't work:
```javascript
import React from 'react';
import Footer from './Footer';
import { createBrowserHistory } from "history";
import { createStore, AnyAction, Store } from 'redux';
import createReducer from '../../../reducers';
import AppProvider from '../../AppProvider';
import { State } from '../../../reducers/domain';

const history = createBrowserHistory();
const store = createStore(
    createReducer(history, {} as State)
) as Store<any, AnyAction>;

export default {
    title: 'Fragments/Footer',
    component: Footer,
    decorators: [
        (story: any) => <AppProvider store={store} history={history}>{story()}</AppProvider>,
    ]
};

export const footer = () => <Footer />;
```
I wont include the full code of the 2 reducers here as it's quite long, but I tried to replace both with the code below and it makes no difference, so I'm fairly confident it's unrelated to the actual contents of those files.
```javascript
export default function(){ return {}; }
```

**System**
Environment Info:

  System:
    OS: macOS 10.15.7
    CPU: (12) x64 Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
  Binaries:
    Node: 12.18.3 - /usr/local/opt/node@12.18.3/bin/node
    npm: 6.14.7 - /usr/local/opt/node@12.18.3/bin/npm
  Browsers:
    Chrome: 87.0.4280.47
    Edge: 86.0.622.63
    Firefox: 81.0.2
    Safari: 14.0
  npmPackages:
    @storybook/addon-a11y: ^6.0.28 => 6.0.28 
    @storybook/react: ^6.0.28 => 6.0.28 

**Additional context**
This did used to work a few months ago, but at some point some code must have been added to my codebase that has broken it. I have also just updated from v5 to v6 in an attempt to fix the issue. When I was running on v5 the whole storybook app would crash (with the same error message), on v6 I do get the stories that aren't importing my reducer showing at least.

**Comment by timothy-bailey-redbox** [comment]
Ok I've actually managed to resolve the issue myself, so I'm going to close this ticket as I believe the issue was relevent to only my codebase. 

But in case it helps anyone else out in the future, the thing that I have changed was to move a refference to the `process.env` variables out of my services files and directly into the reducer file. I don't know why this worked, only that I slowly was narrowing in that the issue seemed to be related to the env variables not being accessable by process of commenting out chucks of code until things started working.
